ðŸ§¬ Strelka2 Somatic Variant Calling Pipeline
This pipeline performs somatic variant calling using Strelka2 from tumor-normal paired whole genome sequencing (WGS) data, using Nextflow and Docker for reproducibility and scalability.

Directory Structure
arduino
Copy
Edit
project/
â”œâ”€â”€ Dockerfile
â”œâ”€â”€ README.md
â”œâ”€â”€ Strelka2_Somatic_variant.nf
â”œâ”€â”€ nextflow.config
â””â”€â”€ data/
    â”œâ”€â”€ patient1/
    â”‚   â”œâ”€â”€ tumor_R1.fastq.gz
    â”‚   â”œâ”€â”€ tumor_R2.fastq.gz
    â”‚   â”œâ”€â”€ normal_R1.fastq.gz
    â”‚   â””â”€â”€ normal_R2.fastq.gz
    â””â”€â”€ ...

ðŸš€ How to Run
1. Install Nextflow:
curl -s https://get.nextflow.io | bash

2. Clone the repository and change into the directory:
git clone https://github.com/lipingzengGitHub/Strelka2_Somatic_varaint.git
cd Strelka2_Somatic_varaint

3. Run the pipeline:
nextflow run Strelka2_Somatic_variant.nf -profile docker

ðŸ”§ Input Requirements:
Each patientâ€™s data must be in a separate folder under data/

Each folder must contain these 4 files:
tumor_R1.fastq.gz
tumor_R2.fastq.gz
normal_R1.fastq.gz
normal_R2.fastq.gz

ðŸ“¦ Docker
Dockerfile is included and used by default. You don't need to install Strelka2 or any dependencies.

ðŸ“¤ Output
Results will be saved under results/ folder per sample

Strelka2 output includes:

somatic.indels.vcf.gz

somatic.snvs.vcf.gz



